### DISCUSSION TITLE
What's the non-gpl and gpl versions?

### DISCUSSION THREAD

**Discussion by b005t3r** [bytedeco/javacpp-presets#1549] [General]
What's the reason behind having both versions instead of just one, with GPL stuff enabled by default? I don't think GPL licensing has any potential drawbacks here, since only the binary is being distributed? Or am I missing something?

**Comment by saudet** [comment]
The GPL license requires that you also publish all of your application code under the GPL. If that's not OK with you, you cannot use GPL code.

That said, I`ll probably get rid of it at some point in time and make the main package GPL, yes, to reduce maintenance costs.

So that means you'll need to publish your application source code under GPL too.

**Comment by b005t3r** [comment]
@saudet 

I think you might be misinterpreting it, you only need to do that, if you include code under GPL in your project, then the project has to be published under GPL and your code has to be made public. But since you only use the binaries (build from the code under GPL), it does NOT transfer to your code.
